wiki/wiki/doctype/wiki_page/wiki_page.py

```diff
@@ -313,39 +321,27 @@ def get_sidebar_items(self):
 
 			wiki_page = frappe.get_doc("Wiki Page", sidebar_item.wiki_page)
 
-			if not wiki_page.allow_guest:
-				permitted = frappe.has_permission(wiki_page.doctype, "read", wiki_page)
-				if not permitted:
-					continue
-
-			if sidebar_item.parent_label not in sidebar:
-				sidebar[sidebar_item.parent_label] = [
-					{
-						"name": wiki_page.name,
-						"type": "Wiki Page",
-						"title": wiki_page.title,
-						"route": wiki_page.route,
-						"group_name": sidebar_item.parent_label,
-					}
-				]
-			else:
-				sidebar[sidebar_item.parent_label] += [
-					{
-						"name": wiki_page.name,
-						"type": "Wiki Page",
-						"title": wiki_page.title,
-						"route": wiki_page.route,
-						"group_name": sidebar_item.parent_label,
-					}
-				]
+			if wiki_page.allow_guest or frappe.has_permission(wiki_page.doctype, "read", wiki_page):
+				page_info = {
+                    "name": wiki_page.name,
+                    "type": "Wiki Page",
+                    "title": wiki_page.title,
+                    "route": wiki_page.route,
+                    "group_name": sidebar_item.parent_label,
+                }
+				if sidebar_item.parent_label not in sidebar:
+					sidebar[sidebar_item.parent_label] = [page_info]
+				else:
+					sidebar[sidebar_item.parent_label].append(page_info)
 
 		return self.get_items(sidebar)
 
 	def get_last_revision(self):
 		last_revision = frappe.db.get_value(
 			"Wiki Page Revision Item", filters={"wiki_page": self.name}, fieldname="parent"
 		)
-		return frappe.get_doc("Wiki Page Revision", last_revision)
+		if frappe.db.exists('Wiki Page Revision', last_revision):
+			return frappe.get_doc("Wiki Page Revision", last_revision)
```

Comment:
`frappe.db.delete_if_exists(...)`

Reply:
Hello @blaggacao, It looks like you have not analysed this code properly. I have added condition to return if exists as before it was throwing **None not found error**. I just handled this not deleting anything.

Thank you!!

Reply:
Yeah, sorry, ignore that :smile: As I said, it came from the phone.